src/asl.ts

```diff
@@ -4871,13 +5016,331 @@ export namespace ASLGraph {
           }),
     };
   };
+
+  /**
+   * Compare any two {@link ASLGraph.Output} values.
+   */
+  export const compareOutputs = (
+    leftOutput: ASLGraph.Output,
+    rightOutput: ASLGraph.Output,
+    operator: ASL.ValueComparisonOperators | "!=" | "!=="
+  ): Condition => {
+    if (
+      ASLGraph.isLiteralValue(leftOutput) &&
+      ASLGraph.isLiteralValue(rightOutput)
+    ) {
+      return ((operator === "==" || operator === "===") &&
+        leftOutput.value === rightOutput.value) ||
+        ((operator === "!=" || operator === "!==") &&
+          leftOutput.value !== rightOutput.value) ||
+        (leftOutput.value !== null &&
+          rightOutput.value !== null &&
+          ((operator === ">" && leftOutput.value > rightOutput.value) ||
+            (operator === "<" && leftOutput.value < rightOutput.value) ||
+            (operator === "<=" && leftOutput.value <= rightOutput.value) ||
+            (operator === ">=" && leftOutput.value >= rightOutput.value)))
```

Comment:
Should we introduce guards instead of these string literal comparisons to match the node guards and to also start developing composed helpers?

Reply:
Not something that changed, I just moved them (new rule: anything that accepts ASLGraph.Output should be in ASLGraph instead of the ASL namespace. ASL are for raw ASL wrappers and output ASL, not ASLGraph).

Not sure if the guards would help too much, seems like they would get quite verbose to cover all operators.

Reply:
Well, I would eventually want to change these to a numeric enum, but OK.

Reply:
I see. Would the guard accept a string operator or would you manually create a guard for all  44(+) operators?

Reply:
All 44.

Reply:
Later is fine :)